### PR TITLE
Create UuidFields value object

### PR DESCRIPTION
**Pull request by nicwortel**

### CODE REVIEW

**Comment by aztech-dev** [comment]
Your work in progress looks nice, but I'm a bit worried about the performance impact of all those assertions in the constructor of UuidFields.

Do you mind providing a benchmark to compare performance before and after your changes ?

File: composer.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     },
     "require": {
         "php": ">=5.6",
+        "beberlei/assert": "^2.5",
```
Comment (GrahamCampbell) [inline composer.json]: This should be in require-dev?

Comment (GrahamCampbell) [inline composer.json]: Oh, right, I see.

Comment (GrahamCampbell) [inline composer.json]: NVm

**Comment by ramsey** [comment]
Checking in on your work in progress. How's this coming?

**Comment by nicwortel** [comment]
Hi, sorry for the delay, I couldn't find the time to finish the PR. I think it's finished now, although I'll have to do some more exploratory testing to be sure everything works.

**Comment by nicwortel** [comment]
@aztech-dev I've created the following file: benchmark.php

```php
<?php

require 'vendor/autoload.php';

for ($i = 0; $i < 1000; $i++) {
    $uuid = \Ramsey\Uuid\Uuid::uuid4();
}
```

Then I execute the following command:

```
time php benchmark.php 
```

On the 4.x branch this gives the following result:

```
real	0m0.048s
user	0m0.040s
sys	0m0.008s
```

On my branch this is the result:

```
real	0m0.089s
user	0m0.080s
sys	0m0.008s
```

**Comment by ramsey** [comment]
On what version of PHP did you run that benchmark?

**Comment by nicwortel** [comment]
PHP 7.0.9-1+deb.sury.org~xenial+1 (cli)

**Comment by nicwortel** [comment]
@ramsey have you decided whether you're gonna merge this yet?

**Comment by ramsey** [comment]
Sorry for the delay. On holiday now; I'll review late next week.

**Comment by nicwortel** [comment]
No problem, enjoy :+1:

**Comment by ramsey** [comment]
I'm still giving some thought to this. I have some ideas about the future structure, and I've been thinking about your suggestion at DPC (I think it was you who suggested this) to introduce separate objects for Uuid1, Uuid4, etc.

Just wanted to let you know this is still on the table. :-)

**Pull request closed without merging**